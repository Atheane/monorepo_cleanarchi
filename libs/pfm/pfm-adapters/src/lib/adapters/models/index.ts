export { AggregationBank } from './bank/Bank';
export { AggregationAccount, SMoneyBankAccount } from './bankAccount/BankAccount';
export * from './transaction';
