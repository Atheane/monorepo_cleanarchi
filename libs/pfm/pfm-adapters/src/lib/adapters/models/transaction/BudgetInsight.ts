enum EventTypeBudgetInsight {
  ACCOUNT_SYNC = 'ACCOUNT_SYNC',
}

export type EventBudgetInsight = {
  date: Date;
  type: EventTypeBudgetInsight;
  data: BankAccount;
  version: number;
};

export type BankAccount = {
  id: number;
  id_connection: number;
  id_user: number;
  id_source: number;
  id_parent?: number;
  number: string;
  webid?: string;
  original_name: string;
  balance: number;
  coming?: number;
  display: boolean;
  last_update?: Date;
  deleted?: Date;
  disabled?: string;
  iban?: string;
  currency: Currency;
  id_type: number;
  bookmarked: number;
  name: string;
  error?: string;
  usage: AccountUsage;
  ownership: string;
  company_name?: string;
  bic?: string;
  coming_balance?: string;
  formatted_balance?: string;
  type: AccountType;
  information?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  loan?: Loan;
  transactions: TransactionBudgetInsight[];
};

export type TransactionBudgetInsight = {
  id: number;
  id_account: number;
  webid?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  application_date?: Date;
  date: Date;
  value: number;
  gross_value?: string;
  original_wording: string;
  simplified_wording: string;
  stemmed_wording?: string;
  wording?: string;
  id_category?: number;
  state?: string;
  date_scraped: Date;
  rdate: Date;
  vdate?: Date;
  bdate?: Date;
  coming: boolean;
  active: boolean;
  id_cluster?: number;
  comment?: string;
  last_update?: Date;
  deleted?: Date;
  original_value?: number;
  original_gross_value?: number;
  original_currency?: Currency;
  commission?: number;
  commission_currency?: Currency;
  country?: string;
  counterparty?: string;
  card?: string;
  category: Category;
  type: TransactionTypeBudgetInsight;
  new: boolean;
  formatted_value: string;
  documents_count: number;
  nopurge: number;
};

export enum TransactionTypeBudgetInsight {
  TRANSFER = 'transfer',
  ORDER = 'order',
  CHECK = 'check',
  DEPOSIT = 'deposit',
  PAYBACK = 'payback',
  WITHDRAWAL = 'withdrawal',
  LOAD_PAYMENT = 'loan_payment',
  BANK = 'bank',
  CARD = 'card',
  DEFERRED_CARD = 'deferred_card',
  SUMMARY_CARD = 'summary_card',
}

export type Currency = {
  id: string;
  symbol: string;
  prefix: boolean;
  crypto: boolean;
  precision: number;
  marketcap?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  datetime?: Date;
  name: string;
};

export type Category = {
  id: number;
  id_parent_category?: number;
  name: string;
  income?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  color: string;
  id_parent_category_in_menu?: number;
  name_displayed: string;
  refundable: boolean;
  id_user?: number;
  id_logo?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  hidden: boolean;
  accountant_account?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
};

export enum AccountType {
  UNKOWN = 1,
  CHECKING = 2,
  SAVINGS = 3,
  DEPOSIT = 4,
  LOAN = 5,
  MARKET = 6,
  JOINT = 7,
  CARD = 8,
  LIFEINSURANCE = 9,
  PEE = 10,
  PERCO = 11,
  ARTICLE83 = 12,
  RSP = 13,
  PEA = 14,
  CAPITALISATION = 15,
  PERP = 16,
  MADELIN = 17,
  MORTGAGE = 18,
  CONSUMERCREDIT = 19,
  REVOLVINGCREDIT = 20,
  PER = 21,
}

export enum AccountUsage {
  PRIV = 'PRIV',
  ORGA = 'ORGA',
  ASSO = 'ASSO',
}

export type Loan = {
  id: number;
  id_account: number;
  contact_name?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  total_amount: number;
  available_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  used_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  subscription_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  maturity_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  next_payment_amount: number;
  next_payment_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  rate?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_left?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_done?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_total?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  last_payment_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  last_payment_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  account_label?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  insurance_label?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  duration?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  type: string;
};
