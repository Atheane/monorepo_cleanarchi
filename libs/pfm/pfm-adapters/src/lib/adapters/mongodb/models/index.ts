export * from './AccountStatement';
export * from './BudgetInsight';
export * from './P2p';
export * from './User';
