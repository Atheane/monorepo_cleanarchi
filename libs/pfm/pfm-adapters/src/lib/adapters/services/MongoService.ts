import * as mongoose from 'mongoose';

export async function initMongooseConnection(url: string): Promise<mongoose.Connection> {
  await mongoose.connect(url, {
    useNewUrlParser: true,
    useFindAndModify: false,
    useUnifiedTopology: true,
    useCreateIndex: true,
  });
  return mongoose.connection;
}

export async function createMongooseConnection(
  url: string,
  poolSize: number,
  maxPoolSize: number,
): Promise<mongoose.Connection> {
  return mongoose.createConnection(url, {
    useNewUrlParser: true,
    useFindAndModify: false,
    useUnifiedTopology: true,
    useCreateIndex: true,
    poolSize: poolSize,
    maxPoolSize: maxPoolSize,
  });
}
