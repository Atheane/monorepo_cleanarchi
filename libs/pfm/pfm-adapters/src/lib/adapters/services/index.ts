export * from './GetAccountStatementService';
export * from './GetBankAccountsService';
export * from './GetTransactionsService';
export * from './MongoService';
