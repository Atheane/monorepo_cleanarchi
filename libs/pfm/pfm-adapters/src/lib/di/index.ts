export * from './PfmKernel';
export * from './DomainDependencies';
