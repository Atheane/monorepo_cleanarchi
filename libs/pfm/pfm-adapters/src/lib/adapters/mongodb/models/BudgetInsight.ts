import { Document, Schema, Connection, Model } from 'mongoose';
import { EventBudgetInsight } from '../../models/transaction/BudgetInsight';

export enum EventType {
  ACCOUNT_SYNC = 'ACCOUNT_SYNC',
}

export type EventDoc = EventBudgetInsight & Document;

export class EventSchema extends Schema {
  constructor() {
    super(
      {
        date: {
          type: String,
          required: true,
        },
        type: {
          type: String,
          required: true,
          index: true,
        },
        data: {
          type: Object,
          required: true,
        },
        version: {
          type: Number,
          required: true,
          index: true,
        },
      },
      {
        versionKey: false,
      },
    );
  }
}

const EventIdentifier = 'budgetinsight_event';

let EventModel;
export const connectEventModel = (connection: Connection): Model<EventDoc> => {
  EventModel = connection.model<EventDoc>(EventIdentifier, new EventSchema());
  return EventModel;
};

export const getEventModel = (): Model<EventDoc> => EventModel;
