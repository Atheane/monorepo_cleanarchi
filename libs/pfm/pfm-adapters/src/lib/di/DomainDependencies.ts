import {
  P2pRepository,
  AccountStatementRepository,
  GetTransactionById,
  GetAllBankAccounts,
  GetTransactionsByAccountId,
  GetAllTransactions,
  CreateP2p,
  GetListAccountStatements,
  GetOneAccountStatement,
  ProcessStatements,
  ProcessMonthlyStatements,
  UserRepository,
} from '@oney/pfm-core';
import {
  BankAccountMapper,
  TransactionMapper,
  AccountStatementMapper,
  BankAccountAggregationMapper,
  TransactionSMoneyMapper,
  UserMapper,
} from '../adapters/mappers';
import { GetTransactionsService } from '../adapters/services/GetTransactionsService';

export interface DomainDependencies {
  createP2p: CreateP2p;
  getAllBankAccounts: GetAllBankAccounts;
  getAllTransactions: GetAllTransactions;
  getTransactionsByAccountId: GetTransactionsByAccountId;
  getListAccountStatements: GetListAccountStatements;
  getOneAccountStatement: GetOneAccountStatement;
  getTransactionById: GetTransactionById;
  p2pRepository: P2pRepository;
  accountStatementRepository: AccountStatementRepository;
  userRepository: UserRepository;
  processStatements: ProcessStatements;
  processMonthlyStatements: ProcessMonthlyStatements;
  mappers: {
    bankAccountAggregationMapper: BankAccountAggregationMapper;
    transactionSMoneyMapper: TransactionSMoneyMapper;
    bankAccountMapper: BankAccountMapper;
    transactionMapper: TransactionMapper;
    accountStatementMapper: AccountStatementMapper;
    userMapper: UserMapper;
  };
  getTransactionService: GetTransactionsService;
}
