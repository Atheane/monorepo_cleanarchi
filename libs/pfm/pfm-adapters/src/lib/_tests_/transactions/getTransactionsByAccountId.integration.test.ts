import * as nock from 'nock';
import * as dateMock from 'jest-date-mock';
import { ConfigService } from '@oney/envs';
import { TransactionError, P2p } from '@oney/pfm-core';
import * as path from 'path';
import { DomainDependencies } from '../../di/DomainDependencies';
import { getEventModel } from '../../adapters/mongodb/models/BudgetInsight';
import { seedEvents } from '../fixtures/transactions/seedEvents';
import { P2pTransactions } from '../fixtures/transactions/P2pTransactions';
import { testConfiguration } from '../fixtures/config/config';
import { initMongooseConnection } from '../../adapters/services/MongoService';
import { setupKernel } from '../fixtures/config/config.kernel';
import { userTestCredentials } from '../fixtures/bankAccounts/userTestCredentials';
import { getAccountStatementModel, getP2pModel } from '../../adapters/mongodb/models';

const nockBack = nock.back;
nockBack.fixtures = path.resolve(`${__dirname}/../fixtures/transactions`);
nockBack.setMode('record');
const envPath = path.resolve(__dirname + '/../env/test.env');

describe('Get getTransactionsByAccountId integration testing', () => {
  let dependencies: DomainDependencies;

  beforeAll(async () => {
    await new ConfigService({ localUri: envPath }).loadEnv();
    const dbConnection = await initMongooseConnection(process.env.MONGO_URL);
    const kernel = await setupKernel(testConfiguration, false, dbConnection);
    await kernel.initIdentityLib();
    kernel.initBlobStorage();
    dependencies = kernel.getDependencies();
    nock.enableNetConnect();
  });

  beforeEach(async () => {
    dateMock.clear();
    await getEventModel().deleteMany({});
    await getP2pModel().deleteMany({});
    await getAccountStatementModel().deleteMany({});
    await getEventModel().insertMany(seedEvents);
  });

  it('should get transactions of a given account for a specific date in ASC order', async () => {
    ////
    const { nockDone } = await nockBack('getTransactionsByAccountIdInASCOrder.json');

    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '8254',
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });
    expect(result.length).toEqual(2);
    expect(
      result.every(
        t =>
          new Date(t.date).getTime() >= new Date('2020-09-01').getTime() &&
          new Date(t.date).getTime() < new Date('2020-09-05').getTime(),
      ),
    ).toBeTruthy();
    expect(new Date(result[0].date).getTime()).toBeLessThanOrEqual(new Date(result[1].date).getTime());
    nockDone();
  });

  it('should get P2P among other transactions for a specific account', async () => {
    const { nockDone } = await nockBack('getP2PTransactionsAmongOthers.json');
    dateMock.advanceTo(new Date('2020-09-01T16:32:29.047Z'));

    const { userId, userToken } = userTestCredentials;
    await Promise.all(P2pTransactions.map(p2p => dependencies.p2pRepository.create(new P2p(p2p))));

    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '1388',
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });

    expect(result).toMatchSnapshot();
    nockDone();
  });

  it('should not get P2P when date not corresponding', async () => {
    const { nockDone } = await nockBack('getP2PTransactionsAmongOthers.json');
    dateMock.advanceTo(new Date('2020-09-06T16:32:29.047Z'));

    const { userId, userToken } = userTestCredentials;
    await Promise.all(P2pTransactions.map(p2p => dependencies.p2pRepository.create(new P2p(p2p))));

    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '1388',
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });

    expect(result).toMatchSnapshot();
    nockDone();
  });

  it('should throw a tInvalid query erro', async () => {
    const { nockDone } = await nockBack('getP2PTransactionsAmongOthers.json');
    dateMock.advanceTo(new Date('2020-09-06T16:32:29.047Z'));

    const { userId, userToken } = userTestCredentials;
    await Promise.all(P2pTransactions.map(p2p => dependencies.p2pRepository.create(new P2p(p2p))));

    const result = dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '1388',
      query: { badParams: 'test' },
    });

    await expect(result).rejects.toThrow(TransactionError.InvalidQuery);
    nockDone();
  });

  it('should throw a transaction not found error', async () => {
    const { nockDone } = await nockBack('transactionNotFound.json');
    const { userId, userToken } = userTestCredentials;
    const result = dependencies.getTransactionById.execute({
      userId,
      userToken,
      transactionId: 'nimp',
    });

    await expect(result).rejects.toThrow(TransactionError.TransactionNotFound);
    nockDone();
  });

  it('should get a list of transactions including P2Ps for a given account id', async () => {
    const { nockDone } = await nockBack('getTransactionsByAccountId.json');

    const { userId, userToken } = userTestCredentials;

    // no http calls needs to be mocked here, because we specify accountId
    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '1388',
    });

    const transaction = result.find(t => t.refId === 'a-9F3q0Ov');
    expect(result.length).toEqual(117);
    expect(transaction.amount.value).toEqual(2);
    expect(transaction.type).toEqual('TRANSFER');

    nockDone();
  });

  it('should get a list of transactions for a given account id', async () => {
    const { nockDone } = await nockBack('getTransactionsByAccountId.json');

    const { userId, userToken } = userTestCredentials;

    // no http calls needs to be mocked here, because we specify accountId
    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '8241',
    });

    const transaction = result.find(t => t.refId === '126855');
    expect(result.length).toEqual(106);
    expect(transaction.amount.value).toEqual(9.9);
    expect(transaction.label).toEqual('PRLV SEPA ParisVilliers');
    expect(transaction.type).toEqual('ORDER');

    nockDone();
  });

  it('should get a list of updated transactions', async () => {
    const { nockDone } = await nockBack('getTransactionsByAccountIdUpdatedList.json');

    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: '8254',
    });

    const transaction = result.find(t => t.refId === '126348');
    expect(result.length).toEqual(2);
    expect(transaction.amount.value).toEqual(76.61);
    expect(transaction.status).toEqual('CLEARED');
    nockDone();
  });

  it('should throw an error bank account not found', async () => {
    const { nockDone } = await nockBack('bankAccountNotFound.json');
    const { userId, userToken } = userTestCredentials;

    const result = dependencies.getTransactionsByAccountId.execute({
      userId,
      userToken,
      accountId: 'nimp',
    });

    await expect(result).rejects.toThrow(TransactionError.AccountNotFound);
    nockDone();
  });

  afterAll(async () => {
    nock.enableNetConnect();
  });
});
