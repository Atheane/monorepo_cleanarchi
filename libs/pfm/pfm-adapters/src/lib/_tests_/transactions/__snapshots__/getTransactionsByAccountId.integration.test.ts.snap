// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get getTransactionsByAccountId integration testing should get P2P among other transactions for a specific account 1`] = `
Array [
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:03.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "a-9F3q0Ov",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:16.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "wZRl5OM2y",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1000,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T08:23:13.512Z,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Jane Doe",
      "iban": "iban_value",
      "id": "456",
    },
    "currency": "EUR",
    "date": 2020-09-03T08:23:13.512Z,
    "direction": "IN",
    "fees": null,
    "isDeposit": false,
    "label": "p2p received",
    "originalAmount": AmountPositive {
      "amount": 1000,
    },
    "refId": "gaerh",
    "rejectionReason": null,
    "status": "CLEARED",
    "tag": "tag_value",
    "type": "P2P",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:20:45.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:20:45.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "Tdle-sdyn",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:21:07.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:21:07.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "vAPKLJrJP",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 6,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:24:42.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:24:42.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 6,
    },
    "refId": "DSPxRpsta",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
]
`;

exports[`Get getTransactionsByAccountId integration testing should not get P2P when date not corresponding 1`] = `
Array [
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:03.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "a-9F3q0Ov",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:16.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "wZRl5OM2y",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1000,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T08:23:13.512Z,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Jane Doe",
      "iban": "iban_value",
      "id": "456",
    },
    "currency": "EUR",
    "date": 2020-09-03T08:23:13.512Z,
    "direction": "IN",
    "fees": null,
    "isDeposit": false,
    "label": "p2p received",
    "originalAmount": AmountPositive {
      "amount": 1000,
    },
    "refId": "gaerh",
    "rejectionReason": null,
    "status": "CLEARED",
    "tag": "tag_value",
    "type": "P2P",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:20:45.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:20:45.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "Tdle-sdyn",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:21:07.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:21:07.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "vAPKLJrJP",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 6,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:24:42.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:24:42.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 6,
    },
    "refId": "DSPxRpsta",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
]
`;
